includes/Core/Admin/Screens.php

```diff
@@ -359,16 +371,62 @@ private function get_screens() {
 			new Screen(
 				self::PREFIX . 'dashboard',
 				array(
-					'title'            => __( 'Dashboard', 'google-site-kit' ),
-					'capability'       => Permissions::VIEW_DASHBOARD,
-					'enqueue_callback' => function ( Assets $assets ) {
+					'title'               => __( 'Dashboard', 'google-site-kit' ),
+					'capability'          => Permissions::VIEW_DASHBOARD,
+					'enqueue_callback'    => function ( Assets $assets ) {
 						if ( $this->context->input()->filter( INPUT_GET, 'permaLink' ) ) {
 							$assets->enqueue_asset( 'googlesitekit-entity-dashboard' );
 						} else {
 							$assets->enqueue_asset( 'googlesitekit-main-dashboard' );
 						}
 					},
-					'render_callback'  => function ( Context $context ) {
+					'initialize_callback' => function ( Context $context ) {
+						$is_view_only = ! $this->authentication->is_authenticated();
+
+						if ( ! $is_view_only ) {
+							$initial_setup_settings      = ( new Initial_Setup_Settings( $this->user_options ) )->get();
+							$is_analytics_setup_complete = $initial_setup_settings['isAnalyticsSetupComplete'];
+
+							if ( false === $is_analytics_setup_complete ) {
+								$is_analytics_connected = $this->modules->is_module_connected( 'analytics-4' );
+
+								if ( $is_analytics_connected ) {
+									wp_safe_redirect(
+										$context->admin_url(
+											'user-input',
+											array(
+												'showProgress' => true,
```

Comment:
Using a boolean `('showProgress' => true)` value is incorrect, which gets converted to `?showProgress=1` in the URL. However, the JavaScript code checks for the string `'true'`:
https://github.com/google/site-kit-wp/blob/3d1320f43907bd2ff01073cc89cc17929ce20172/assets/js/components/setup/ModuleSetup.js#L118-L121
This means the progress indicator won't display because `"1" === 'true'` evaluates to `false`. We should change to `'showProgress' => 'true'` (string) to match the pattern used in `Analytics_4.php`:
https://github.com/google/site-kit-wp/blob/3d1320f43907bd2ff01073cc89cc17929ce20172/includes/Modules/Analytics_4.php#L1003-L1010
```suggestion
												'showProgress' => 'true',
```